src/shogun/mathematics/graph/ops/shogun/Add.h

```diff
@@ -55,7 +56,35 @@ namespace shogun
 					if (CPU_arch->has_avx512f())
 						add_kernel_implementation_avx512f<T>(input1, input2, output, size);
 					else if (CPU_arch->has_avx())
-						add_kernel_implementation_avx<T>(input1, input2, output, size);
+					{
+						size_t i = 0;
+						if (size > 32/sizeof(T))
+						{
+							size_t remainder = size % (32/sizeof(T));
+							for (;i<size-remainder; i+=32/sizeof(T))
```

Comment:
`32` because it's the width in bytes of a YMM avx register

Reply:
probably good to put a comment in the source for that?

Reply:
this logic would be moved to an executor, this is just PoC for now :) but yes, I will also replace these things with some Eigen definitions